File: 2000 98 & ME/QuickStart/howto/samples/Services/DirectoryServices/adread/VB/adread.vb
imports System
imports System.DirectoryServices
        
public class ADRead
        public shared sub Main()
        if(Environment.GetCommandLineArgs().Length<>2)
                Console.WriteLine("Usage: " & Environment.GetCommandLineArgs()(0) & " <ad_path>")
                exit sub
        end if

        Dim objDirEnt As DirectoryEntry = new DirectoryEntry(Environment.GetCommandLineArgs(1))
        Console.WriteLine("Name            = " & objDirEnt.Name)
        Console.WriteLine("Path            = " & objDirEnt.Path)
        Console.WriteLine("SchemaClassName = " & objDirEnt.SchemaClassName)
        Console.WriteLine("")
        Console.WriteLine("Properties:")

        Dim tab As string = "    "
        Dim objP as DirectoryEntryProperty
        Dim objValue as object

        for each objP in objDirEnt.Properties
            Console.Write(tab & objP.Name & " = ")
            Console.WriteLine("")
            for each objValue in objP.Values
                    Console.WriteLine(tab & tab & objValue.ToString())
            next objValue
        next objP
    end sub             
end class

File: 2000 98 & ME/QuickStart/howto/samples/Services/DirectoryServices/adread/adread.cs
using System;
using System.DirectoryServices;
	
public class ADRead
{
    public static void Main(String[] args) 
	{
		if(args.Length!=1)
        {
         	Console.WriteLine("Usage: " + Environment.GetCommandLineArgs()[0] + " <ad_path>");
         	return;
        }

        DirectoryEntry objDirEnt=new DirectoryEntry(args[0]);
        Console.WriteLine("Name            = " + objDirEnt.Name);
        Console.WriteLine("Path            = " + objDirEnt.Path);
        Console.WriteLine("SchemaClassName = " + objDirEnt.SchemaClassName);
        Console.WriteLine("");
        Console.WriteLine("Properties:");
            	
		string tab = "    ";
        foreach(DirectoryEntryProperty objP in objDirEnt.Properties)
        {
            Console.Write(tab + objP.Name + " = ");
            Console.WriteLine("");	
            foreach(Object objValue in objP.Values) 
                Console.WriteLine(tab + tab + objValue);
        }
    }		
}